vividus-plugin-winrm/src/main/java/org/vividus/winrm/WinRmSteps.java

```diff
@@ -0,0 +1,118 @@
+     */
+    @When("I execute batch command `$command` on server `$server` using WinRM and save result as JSON to $scopes "
+            + "variable `$variableName`")
+    public void executeBatchCommand(String command, String server, Set<VariableScope> scopes, String variableName)
+    {
+        executeCommand(server, winRmTool -> winRmTool.executeCommand(command), scopes, variableName);
+    }
+
+    /**
+     * Executes a PowerShell command. A new session is created on the destination host for each step invocation.
+     *
+     * @param command      The PowerShell command.
+     * @param server       The WinRM server key matching one configured in the properties.
+     * @param scopes       The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of the variable
+     *                     scopes.<br>
+     *                     <i>Available scopes:</i>
+     *                     <ul>
+     *                     <li><b>STEP</b> - the variable will be available only within the step,
+     *                     <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                     <li><b>STORY</b> - the variable will be available within the whole story,
+     *                     <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                     </ul>
+     * @param variableName The variable name to store the command execution results.
+     */
+    @When("I execute PowerShell command `$command` on server `$server` using WinRM and save result as JSON to $scopes "
+            + "variable `$variableName`")
+    public void executePowerShellCommand(String command, String server, Set<VariableScope> scopes, String variableName)
+    {
+        executeCommand(server, winRmTool -> winRmTool.executePs(command), scopes, variableName);
+    }
+
+    private void executeCommand(String server, Function<WinRmTool, WinRmToolResponse> executor,
+            Set<VariableScope> scopes, String variableName)
+    {
+        ServerConfiguration serverConfiguration = serverConfigurations.get(server,
+                "WinRM server connection with key '%s' is not configured in properties", server);
+
+        WinRmTool.Builder winRmToolBuilder = WinRmTool.Builder.builder(serverConfiguration.getAddress(),
+                serverConfiguration.getUsername(), serverConfiguration.getPassword());
+        winRmToolBuilder.disableCertificateChecks(serverConfiguration.isDisableCertificateChecks());
+        ofNullable(serverConfiguration.getAuthenticationScheme()).ifPresent(winRmToolBuilder::authenticationScheme);
+
+        WinRmClientContext context = WinRmClientContext.newInstance();
+        try
+        {
+            WinRmTool winRmTool = winRmToolBuilder.context(context).build();
+            WinRmToolResponse response = executor.apply(winRmTool);
+            variableContext.putVariable(scopes, variableName, jsonUtils.toJson(response));
```

Comment:
Why you picked JSON ?
Why not follow same approach as SSH steps?

Reply:
> Why you picked JSON ?

it could be XML by nature of WinRM, but who knows XML nowadays

> Why not follow same approach as SSH steps?

as it's turned out, users don't know which variables are available and they don't dive deep into report logs

Reply:
But this JSON doesn't have business  value isn't it?  As far as I know, usually tester validates output of the command via regular expression, won't it overcomplicate validations for no reason?